### PR TITLE
feat: Add lane detection training scripts and ADAS module

### LINKED ISSUE #680: Lane-Detection Training Script and Configuration
## 1 - Short Description
Configure and train a YOLOv11n-seg model for lane detection on the collected DrivaPi dataset. This covers environment setup, hyperparameter configuration, training execution, and evaluation of model performance.

## 2 - Tasks
- [ ] Set up training environment and install dependencies (Ultralytics, CUDA, etc.), configure YOLO11 training parameters (epochs, batch size, image size, augmentation), run training, evaluate on validation and test sets, tune hyperparameters, and export best checkpoint for compilation.

## 3 - Acceptance Criteria
- [ ] A short comparison of at least two hyperparameter configurations is documented, including mAP and mask quality results for each.
- [ ] Model is trained and evaluated on the lane detection dataset, with the best checkpoint saved and validated on at least 20 held-out samples.

## 4 - Notes / Implementation Details
- Base model: `yolo11n-seg.pt` from Ultralytics
- Training command: `yolo segment train data=lanes.yaml model=yolo11n-seg.pt epochs=100 imgsz=640`
- Primary hyperparameters to tune: `lr0`, `lrf`, `batch`, `imgsz`
- Suggested training approach:
  - Start with default Ultralytics hyperparameters as baseline
  - Enable mosaic and HSV augmentation to improve robustness across track conditions
  - Run at least two configurations and compare mAP50 and mask loss
- Suggested evidence to attach in the issue/PR:
  - Training config file (`lanes.yaml` + hyperparameter overrides)
  - Loss and mAP curves (TensorBoard or W&B export)
  - Visual sample of segmentation masks on held-out test images
  - Brief conclusion on whether the checkpoint is ready for export and deployment

### PR DESCRIPTION
**Pull request by AfonsoMota-132**
# Pull Request

**Related issue(s):** closes #680

## Type of change
- [x] feat: A new feature
- [ ] fix: A bug fix
- [ ] docs: Documentation only changes
- [ ] style: Formatting, missing semicolons, etc (no code change)
- [ ] refactor: Code change that neither fixes a bug nor adds a feature
- [ ] perf: A code change that improves performance
- [ ] test: Adding or updating tests
- [ ] ci: CI configuration changes
- [ ] chore: Maintenance tasks
- [ ] spike: Investigation / research

## Summary
Added comprehensive lane detection training scripts and configuration to `ADAS/lane-detection/scripts/`. Includes DDP training with Ultralytics, model conversion to ONNX, Hailo quantization/compilation pipeline, and Docker support for Hailo AI training environment. Also added general README for lane detection module and supporting utilities for calibration and data conversion.

## How to test / Validation
1. Review the new training script structure in `ADAS/lane-detection/scripts/`
2. Check the documentation in `ADAS/lane-detection/README.md` and `TRAINING_QUICK_REFERENCE.md`
3. Verify all Python scripts follow project naming conventions
4. Validate Docker build scripts and Hailo AI configuration

## Checklist
- [x] I have run the project tests locally and they pass
- [x] CI checks are green for this branch (or I will fix any failures)
- [x] I have added or updated tests where applicable
- [x] I have updated documentation (if applicable)
- [x] I have added/updated any migration notes (if database or protocol changes)
- [x] I have requested appropriate reviewers and added labels if needed
- [x] This PR contains no secrets or sensitive data

## Risks and backward compatibility
None

## Related / dependent PRs
None

---
**Approval:** Requires a minimum of **2** approvals.
**Action:** The feature branch **MUST** be deleted upon successful merge.

### CODE REVIEW

File: ADAS/lane-detection/README.md
Comment (Hugofslopes) [inline ADAS/lane-detection/README.md:1]: The contributing and license parts make sense?

**Review by Hugofslopes — CHANGES_REQUESTED** [review]
Please check the comment

**Review by Hugofslopes — APPROVED** [review]
All good. Great job, easy to understand even for someone that does not know nothing about python (that is my case)

File: ADAS/lane-detection/train/config.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        
+    Returns:
+        argparse.Namespace: Updated arguments with config values
+    """
+    if config_dict:
+        # Update args with config values (keys can use hyphens in YAML)
+        for key, value in config_dict.items():
+            # Convert hyphens to underscores for Python attribute names
+            key = key.replace("-", "_")
+            if hasattr(args, key):
```
Comment (melaniereis) [inline ADAS/lane-detection/train/config.py]: YAML values currently overwrite CLI args because this runs after parse_arguments(). This contradicts the documented precedence where CLI should take priority (e.g., --config ... --epochs 200 may still end up using YAML epochs). Shouldn't we ensure CLI overrides YAML as documented?

File: ADAS/lane-detection/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- **Training Pipeline**: Simplified single-PC training with configuration management
+- **Metrics & Evaluation**: Automatic calculation of training metrics and model performance tracking
+- **Reproducibility**: YAML-based configuration for consistent, repeatable training runs
+
+## Quick Start
+
+Navigate to the training directory and run training:
+
+```bash
+cd ADAS/lane-detection/scripts/train
```
Comment (melaniereis) [inline ADAS/lane-detection/README.md]: README/guide mention ADAS/lane-detection/scripts/train, but files are under ADAS/lane-detection/train.
Also link paths to solo_train.md should be updated accordingly.

File: ADAS/lane-detection/train/trainer.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    if int(local_rank) != 0:
+        return
+    
+    save_dir = os.path.join(args.path, args.name)
+    os.makedirs(save_dir, exist_ok=True)
+    
+    # Build results dictionary with all training information
+    results_dict = {
+        "training_complete": True,
+        "final_metrics": results.results_dict if hasattr(results, 'results_dict') else {},
```
Comment (melaniereis) [inline ADAS/lane-detection/train/trainer.py]: Key mismatch: here we read results_dict, but in trainer.py we save metrics under final_metrics. Can we align these keys so metric comparison works correctly?

**Review by melaniereis — CHANGES_REQUESTED** [review]

**Review by melaniereis — APPROVED** [review]
Approved!!!
